### PR TITLE
Implement remember me and default to true

### LINKED ISSUE #4106: Show both login and password at login form
It's very-very annoying when I cannot use my auto-fill in browser because "password" field appears only after one click to "Continue". Thus it causes few extra clicks every time.

image

Please show both textboxes at the same time. If there is any kind of SSO support, then better *hide* password textbox when one enters SSO-related email.

In combination with non-working "Remember me" (#210) it creates a lot of inconvenience

### LINKED ISSUE #4107: "Remember me" doesn't work (or too short?)
It seems like "Remember me" doesn't really work as expected. Even if I use it, next time I see login form again. Not sure, maybe there is some expiration time for cookies which is too short (I login to Couchers somewhere between weekly-monthly)

Please check it and make sure it actually works and remembers the user for some meaningful amount of time (I'd suggest one year)

### LINKED ISSUE #4108: Enable "Remember me" by default
I think it would be beneficial for all users to have "remember me" checkbox set to "Enabled" all the time, this should be the most used functionality.

Currently it requires extra click during every login which is inconvenient

### LINKED ISSUE #4175: Remember me switch is just for decoration
IMO we should either:

- get rid of it completely; or

- Connect it properly, turn it on by default, change the backend behaviour so that when it is not selected, the session expires after the session

### PR DESCRIPTION
**Pull request by aapeliv**
Closes #4107, closes #4108, closes #4175, closes #4106

**Web frontend checklist**
- [x] Formatted my code with `make format`
- [x] There are no warnings from `make lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
This cannot be merged before the password reset flow is fixed in the frontend! There is currently no way to reset password if it's like this since there's no way to do magic link login!

File: app/web/features/auth/login/LoginForm.tsx [hunk 1]
```diff
@@ -87,63 +74,59 @@ export default function LoginForm() {
...
-            control={<Switch size="small" />}
-            label={t("auth:login_page.form.remember_me")}
+          <Controller
+            control={control}
+            name="rememberDevice"
+            defaultValue={true}
+            render={({ onChange, value }) => (
+              <FormControlLabel
+                className={classes.rememberSwitch}
+                value="rememberDevice"
```
Comment (darrenvong) [inline app/web/features/auth/login/LoginForm.tsx]:
```suggestion
```
Would this still work if it's left out? It's a bit confusing to have a value of `rememberDevice`when it's supposed to be the field's name

Comment (aapeliv) [reply app/web/features/auth/login/LoginForm.tsx]: Let's see!

**Comment by aapeliv** [comment]
This PR is back to being all about "remember me", other stuff moved to #4254

**Pull request merged**